File: python/bishop/point.py
class Point:
    def __init__(self, lineal):
        self.x = self.calcX(lineal)
        self.y = self.calcY(lineal)

    def __eq__(self, other):
        return self.x == other.x and self.y == other.y

    def calcX(self, position):
        return (position - 1) // 8

    def calcY(self, position):
        return (position - 1) % 8

File: ruby/lib/point.rb
class Point
  attr_reader :x, :y

  def initialize(position)
    @x = calcX(position)
    @y = calcY(position)
  end

  def calcX(position)
    (position - 1) / 8
  end

  def calcY(position)
    (position - 1) % 8
  end

  def ==(other_point)
    self.x == other_point.x && self.y == other_point.y
  end
end
